core/src/apps/common/passphrase.py

```diff
@@ -17,7 +17,7 @@ def is_enabled() -> bool:
 
 async def get_passphrase(msg: ThpCreateNewSession) -> str:
     if not is_enabled():
-        return ""
+        raise DataError("Passphrase protection is disabled")
```

Comment:
this should be `if not is_enabled() and msg.passphrase != ""`

Reply:
(or probably: `if not is_enabled -> if msg.passphrase != "" then raise; else return ""` )